### ISSUE TITLE
Local Database (with Credentials) is lost after HAOS update to 12.4

### ISSUE THREAD

**Issue by FrankGiesecke** [hassio-addons/addon-emqx#100]
# Problem/Motivation

I've updated my HAOS to version 12.4 - all docker containers are restarted.
After the update, all connections to EMQX where canceled with message "invalid login"
I could not log into the dashboard using my admin password.
But the default password works.
Im EMQX I could see a configured Built-In Database but without any users. 

## Expected behavior

The Built-In Database should survive OS updates.

## Actual behavior

The admin password and the Built-In Database are gone

**Comment by whc2001** [comment]
I just did a full backup migration to a new server, and seems like the data store of this addon is not backed up at all. After restore it's just like a new instance.

**Comment by allout58** [comment]
This is critical and has bitten me too

**Comment by FrankGiesecke** [comment]
Are there any update to this topic?

**Comment by sinclairpaul** [comment]
In summary, if the hostname changes then this would be an issue as the config appears to be placed within a directory based on it.

This explains migrations to other systems, however it doesn't explain why an OS upgrade would cause the issue.

**Comment by whc2001** [comment]
> In summary, if the hostname changes then this would be an issue as the config appears to be placed within a directory based on it.
> 
> This explains migrations to other systems, however it doesn't explain why an OS upgrade would cause the issue.

I later also found that, but trying to move the old config files info the directory of new hostname is not working for me somehow. In the end `emqx ctl data import/export` worked for me.

**Comment by FrankGiesecke** [comment]
> In summary, if the hostname changes then this would be an issue as the config appears to be placed within a directory based on it.
> 
> This explains migrations to other systems, however it doesn't explain why an OS upgrade would cause the issue.

Interesting and good to know.
But in my case, the hostname didn't change.

**Comment by whc2001** [comment]
> > In summary, if the hostname changes then this would be an issue as the config appears to be placed within a directory based on it.
> > This explains migrations to other systems, however it doesn't explain why an OS upgrade would cause the issue.
> 
> Interesting and good to know. But in my case, the hostname didn't change.
> 
> Is there a simple way to export/import the data from the UI to avoid dealing with docker commands in HAOS?

I don't think so. The EMQX WebUI does not seem to provide this functionality.

**Comment by whc2001** [comment]
Not stale

**Comment by allout58** [comment]
Not stale

**Comment by allout58** [comment]
Still needs to be addressed

**Comment by allout58** [comment]
Can we make it so this doesn't keep going stale? Data loss is kinda a critical in my opinion...

**Comment by whc2001** [comment]
1

**Comment by lcnittl** [comment]
Since there were no changes made to address this, this issue is not stale

**Comment by whc2001** [comment]
1

**Comment by allout58** [comment]
Not stale

**Comment by allout58** [comment]
Not solved
